test/WebJobs.Script.Tests.Integration/DotNetEndToEndTests.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Script.Tests.Properties;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Script.Tests
+{
+    public class DotNetEndToEndTests : EndToEndTestsBase<DotNetEndToEndTests.TestFixture>
+    {
+        public DotNetEndToEndTests(TestFixture fixture) : base(fixture)
+        {
+        }
+
+        [Fact]
+        public async Task Invoking_DotNetFunction()
+        {
+            var request = new HttpRequestMessage(HttpMethod.Get, "http://functions/myfunc");
+            Dictionary<string, object> arguments = new Dictionary<string, object>()
+            {
+                { "req", request }
+            };
+
+            await Fixture.Host.CallAsync("DotNetFunction", arguments);
+
+            HttpResponseMessage response = (HttpResponseMessage)request.Properties[ScriptConstants.AzureFunctionsHttpResponseKey];
+
+            Assert.Equal("Hello from .NET", await response.Content.ReadAsStringAsync());
+        }
+
+        public class TestFixture : EndToEndTestFixture
+        {
+            private const string ScriptRoot = @"TestScripts\DotNet";
+            private static readonly string FunctionPath;
+
+            static TestFixture()
+            {
+                FunctionPath = Path.Combine(ScriptRoot, "DotNetFunction");
+                CreateFunctionAssembly();
+            }
+
+            public TestFixture() : base(ScriptRoot, "dotnet")
+            {
+            }
+
+            public override void Dispose()
+            {
+                base.Dispose();
+
+                if (Directory.Exists(FunctionPath))
+                {
+                    Directory.Delete(FunctionPath, true);
+                }
+            }
+
+            private static void CreateFunctionAssembly()
+            {
+                Directory.CreateDirectory(FunctionPath);
+
+                string primaryReferenceSource = @"
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host;
+
+namespace TestFunction
+{
+    public class Function
```

Comment:
might as well colocate the source and the json - either both in source, or both in resx